lib/ansible/module_utils/docker/common.py

```diff
@@ -639,10 +639,12 @@ def _image_lookup(self, name, tag):
         images = response
         if tag:
             lookup = "%s:%s" % (name, tag)
+            lookup_digest = "%s@%s" % (name, tag)
             images = []
             for image in response:
                 tags = image.get('RepoTags')
-                if tags and lookup in tags:
+                digests = image.get('RepoDigests')
+                if tags and lookup in tags or digests and lookup_digest in digests:
```

Comment:
This way it's easier to read:
```suggestion
                if (tags and lookup in tags) or (digests and lookup_digest in digests):
```